### ISSUE TITLE
Using the attester and verifier on two different devices

### ISSUE THREAD

**Issue by Asiderr** [Fraunhofer-SIT/charra#15]
Hi,

We are trying to use an attester and a verifier on two different devices.
We built two docker containers, and we've changed the IP address in the verifier script.
The attestation process is failing in the following way:

Attester:
```
bob@8f53b5bb7eb5:~/charra$ ./bin/verifier 
18:51:37 INFO  src/verifier.c:84: [verifier] Starting up.
18:51:37 INFO  src/verifier.c:93: [verifier] Initializing CoAP endpoint.
18:51:37 INFO  src/verifier.c:104: [verifier] Registering CoAP resource handlers.
18:51:37 INFO  src/verifier.c:108: [verifier] Creating new attestation request.
18:51:37 INFO  src/verifier.c:131: [verifier] Marshaling attestation request data.
18:51:37 INFO  src/verifier.c:145: [verifier] Adding attestation request data to CoAP PDU.
18:51:37 INFO  src/verifier.c:153: [verifier] Sending CoAP message.
18:51:37 INFO  src/verifier.c:262: [verifier] Resource 'attest': Received message.
18:51:37 INFO  src/verifier.c:275: [verifier] Received data of length 645.
18:51:37 INFO  src/verifier.c:279: [verifier] Parsing received CBOR data.
18:51:37 INFO  src/verifier.c:302: [verifier] Starting verification.
18:51:37 INFO  src/verifier.c:312: [verifier] Loading TPM key.
18:51:37 INFO  src/core/charra_key_mgr.c:36: Loading key "PK.RSA.default".
18:51:38 INFO  src/util/tpm2_util.c:128: Primary Key created successfully.
18:51:38 INFO  src/verifier.c:318: [verifier] Loading of TPM key successful.
18:51:38 INFO  src/verifier.c:322: [verifier] Preparing TPM quote verification.
18:51:38 INFO  src/verifier.c:331: [verifier] Verifying TPM Quote signature.
WARNING:esys:src/tss2-esys/api/Esys_VerifySignature.c:302:Esys_VerifySignature_Finish() Received TPM Error 
ERROR:esys:src/tss2-esys/api/Esys_VerifySignature.c:103:Esys_VerifySignature() Esys Finish ErrorCode (0x000002db) 
18:51:38 ERROR src/util/tpm2_util.c:324: Esys_VerifySignature
18:51:38 INFO  src/verifier.c:358: [verifier] +----------------------------+
18:51:38 INFO  src/verifier.c:359: [verifier] |     ATTESTATION FAILED     |
18:51:38 INFO  src/verifier.c:360: [verifier] +----------------------------+
```

Verifier:
```
bob@4c31f55d60d0:~/charra$ ./bin/attester 
18:51:34 INFO  src/attester.c:72: [attester] Starting up.
18:51:34 INFO  src/attester.c:81: [attester] Initializing CoAP endpoint.
18:51:34 INFO  src/attester.c:93: [attester] Registering CoAP resources.
18:51:34 INFO  src/util/coap_util.c:52: Adding CoAP FETCH resource 'attest'.
18:51:34 INFO  src/attester.c:100: [attester] Waiting for connections.
18:51:37 INFO  src/attester.c:128: [attester] Resource 'attest': Received message.
18:51:37 INFO  src/attester.c:139: [attester] Received data of length 75.
18:51:37 INFO  src/attester.c:143: [attester] Parsing received CBOR data.
18:51:37 INFO  src/attester.c:153: [attester] Preparing TPM quote data.
18:51:37 INFO  src/attester.c:182: [attester] Loading TPM key.
18:51:37 INFO  src/core/charra_key_mgr.c:36: Loading key "PK.RSA.default".
18:51:37 INFO  src/util/tpm2_util.c:128: Primary Key created successfully.
18:51:37 INFO  src/attester.c:190: [attester] Do TPM Quote.
18:51:37 INFO  src/attester.c:198: [attester] TPM Quote successful.
18:51:37 INFO  src/attester.c:204: [attester] Preparing response.
18:51:37 INFO  src/attester.c:212: [attester] Marshaling response.
18:51:37 INFO  src/attester.c:220: [attester] Adding marshaled data to CoAP response PDU and send it.
18:51:37 INFO  src/attester.c:100: [attester] Waiting for connections.
```

README says that there are plans for extended verification of claims
with known-good values. Are there any plans that will come up in the near future?
Do you already have an idea of what it should look like? We (@3mdeb team) are
interested in helping with the implementation.

**Comment by eckelmeckel** [comment]
Hi @Asiderr,

thanks for pointing out that problem. At the moment, verification uses the same TPM (simulator) to verify the signature. That could be the problem when using two different devices, containers, or simulators. In the future verification should be done using mbed-tls.

Checking against known-good values was introduced recently and fixed by pull request #19. Thanks for raising the issue of missing files. :-)

**Comment by eckelmeckel** [comment]
Hi Norbert,

as your PR is now merged, this issue is resolved. Closing it.

Thanks,
Michael
